### PR TITLE
Make Interactor an Area3D node.

### PR DESCRIPTION
**Pull request by Zach777**
Make Interactor use Area3D to check for possible interactions.
Also place Interactor files in SceneComponent/Utilites/Interactor instead of being in Legacy.

Resolves issue #326

### CODE REVIEW

**Comment by Zach777** [comment]
> Why not make it a Component, like other `AComponent` types?
> This would make it implementation agnostic as stated in line 54.
> 
> Should also give it full access to `Entity` APIs in it's parent entity

Not entirely sure what you mean by Entity. By implementation agnostic I meant that Interactor can be placed in anything and is not reliant on the player pressing a certain button. Whatever has the Interactor knows that it can interact with things.

**Comment by Reousa** [comment]
> Not entirely sure what you mean by Entity. 

Refer to [Entities](http://docs.moonwards.com/dev/moonwards-composition), `AComponent`s *are* implementation agnostic :P

Having it implemented this way would open up options for better networked implementations as well.

**Comment by Zach777** [comment]
Please look at the PlayerInteractor component and let me know if that was what you were intending.

File: SceneComponent/Actor/PlayerInteractor.gd [hunk 1]
```diff
@@ -0,0 +1,12 @@
+extends AComponent
+
+export var user : NodePath = get_path()
```
Comment (Reousa) [inline SceneComponent/Actor/PlayerInteractor.gd:3]: `AComponent` already contains an implementation to get the entity that owns it, `self.entity`.
It just needs to be a direct child of the `entity`.

File: SceneComponent/Actor/PlayerInteractor.gd [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+
+#This function is required by AComponent.
+func _init().("PlayerInteractor", true) -> void :
+	pass
+
+#Make Interactor have my user variable as it's user.
+func _ready() -> void :
+	$Interactor.user = user
```
Comment (Reousa) [inline SceneComponent/Actor/PlayerInteractor.gd:12]: ditto

File: SceneComponent/Utilities/Interactor/Interactor.gd [hunk 3]
```diff
@@ -0,0 +1,57 @@
+extends Area
+
+"""
+	Gives a node the ability to interact with interactables.
+"""
+
+#This is what I pass as the interactor.
+export var user : NodePath = get_path()
```
Comment (Reousa) [inline SceneComponent/Utilities/Interactor/Interactor.gd:8]: ditto

File: SceneComponent/Utilities/Interactor/Interactor.gd [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+signal interact_possible(string_describing_potential_interact)
+signal interact_became_impossible()
+
+#What interactable I am closest to and can interact with.
+var collider : Area = null
+#This is how I will not spam a signal when I have potential interacts.
+var previous_collider : Area = null
+
+
+#warning-ignore:unused_argument
```
Comment (Reousa) [inline SceneComponent/Utilities/Interactor/Interactor.gd:19]: Could simply rename `delta` to `_delta`, makes it clearer that it's unused & eliminates the need for a `warning-ignore`.

File: SceneComponent/Actor/PlayerInteractor.gd (hunk 2 above)
Comment (Zach777) [reply SceneComponent/Actor/PlayerInteractor.gd:12]: In this case the Interactor needs to know what to pass Interactables.

File: SceneComponent/Utilities/Interactor/Interactor.gd (hunk 4 above)
Comment (Zach777) [reply SceneComponent/Utilities/Interactor/Interactor.gd:19]: Will do. I did not know that was possible lol.

File: SceneComponent/Utilities/Interactor/Interactor.gd (hunk 3 above)
Comment (Zach777) [reply SceneComponent/Utilities/Interactor/Interactor.gd:8]: If we do this, Interactor will always require a parent AComponent to work.

Comment (Zach777) [reply SceneComponent/Utilities/Interactor/Interactor.gd:8]: Would it make more sense to just pass a string that says who is calling it? Or possibly an Array of strings? That way Interactables can just check what is calling it through the string/array of strings passed.

**Pull request merged**